lua/swenv/api.lua

```diff
@@ -148,15 +149,33 @@ local get_pyenv_base_path = function()
   end
 end
 
+local to_set = function (some_list)
+  local set = {} 
+  for _, key in ipair(some_list) do
+    set[key] = true
+  end
+  return set
+end
+
 M.get_venvs = function(venvs_path)
   local venvs = {}
   vim.list_extend(venvs, get_venvs_for(venvs_path, 'venv'))
-  vim.list_extend(venvs, get_venvs_for(get_pixi_base_path(), 'pixi'))
-  vim.list_extend(venvs, get_venvs_for(get_conda_base_path(), 'conda'))
-  vim.list_extend(venvs, get_conda_base_env())
-  vim.list_extend(venvs, get_venvs_for(get_micromamba_base_path(), 'micromamba'))
-  vim.list_extend(venvs, get_venvs_for(get_pyenv_base_path(), 'pyenv'))
-  vim.list_extend(venvs, get_venvs_for(get_pyenv_base_path(), 'pyenv', { only_dirs = false }))
+
+  local ignore_envs = to_set(settings.ignore_envs_groups)
+  if ignore_envs and not ignore_envs['conda'] then
+    vim.list_extend(venvs, get_venvs_for(get_conda_base_path(), 'conda'))
```

Comment:
Just to structure things a bit more, what do you think about moving this to a module level table such as:
```lua
local venvs_getters = {
  conda = function()
    local venvs = {}
    vim.list_extend(venvs, get_venvs_for(get_conda_base_path(), 'conda'))
    vim.list_extend(venvs, get_conda_base_env())
    return venvs
  end,
  pixi = function()
    return get_venvs_for(get_pixi_base_path(), 'pixi')
  end,
  micromamba = function()
    return get_venvs_for(get_micromamba_base_path(), 'micromamba')
  end,
  pyenv = function()
    local venvs = {}
    vim.list_extend(venvs, get_venvs_for(get_pyenv_base_path(), 'pyenv'))
    vim.list_extend(venvs, get_venvs_for(get_pyenv_base_path(), 'pyenv', { only_dirs = false }))
    return venvs
  end,
}
```
and then this main function could be something like:
```lua
for name, getter in pairs(venvs_getters) do
  if not ignore_envs[name] then
    vim.list_extend(venvs, getter())
  end
end
```

Reply:
May be, may be... Extract this table and venv_getters file ("venv_profiles"), ore some thing other?

Reply:
Sorry, not fully sure what you're asking?